lib/DOMMatrix.js

```diff
@@ -445,9 +445,132 @@ DOMMatrix.prototype.flipY = function () {
+          m[12] * m[7] * m[9];
+
+  inv[12] = -m[4]  * m[9] * m[14] + 
+          m[4]  * m[10] * m[13] +
+          m[8]  * m[5] * m[14] - 
+          m[8]  * m[6] * m[13] - 
+          m[12] * m[5] * m[10] + 
+          m[12] * m[6] * m[9];
+
+  inv[1] = -m[1]  * m[10] * m[15] + 
+          m[1]  * m[11] * m[14] + 
+          m[9]  * m[2] * m[15] - 
+          m[9]  * m[3] * m[14] - 
+          m[13] * m[2] * m[11] + 
+          m[13] * m[3] * m[10];
+
+  inv[5] = m[0]  * m[10] * m[15] - 
+          m[0]  * m[11] * m[14] - 
+          m[8]  * m[2] * m[15] + 
+          m[8]  * m[3] * m[14] + 
+          m[12] * m[2] * m[11] - 
+          m[12] * m[3] * m[10];
+
+  inv[9] = -m[0]  * m[9] * m[15] + 
+          m[0]  * m[11] * m[13] + 
+          m[8]  * m[1] * m[15] - 
+          m[8]  * m[3] * m[13] - 
+          m[12] * m[1] * m[11] + 
+          m[12] * m[3] * m[9];
+
+  inv[13] = m[0]  * m[9] * m[14] - 
+          m[0]  * m[10] * m[13] - 
+          m[8]  * m[1] * m[14] + 
+          m[8]  * m[2] * m[13] + 
+          m[12] * m[1] * m[10] - 
+          m[12] * m[2] * m[9];
+
+  inv[2] = m[1]  * m[6] * m[15] - 
+          m[1]  * m[7] * m[14] - 
+          m[5]  * m[2] * m[15] + 
+          m[5]  * m[3] * m[14] + 
+          m[13] * m[2] * m[7] - 
+          m[13] * m[3] * m[6];
+
+  inv[6] = -m[0]  * m[6] * m[15] + 
+          m[0]  * m[7] * m[14] + 
+          m[4]  * m[2] * m[15] - 
+          m[4]  * m[3] * m[14] - 
+          m[12] * m[2] * m[7] + 
+          m[12] * m[3] * m[6];
+
+  inv[10] = m[0]  * m[5] * m[15] - 
+          m[0]  * m[7] * m[13] - 
+          m[4]  * m[1] * m[15] + 
+          m[4]  * m[3] * m[13] + 
+          m[12] * m[1] * m[7] - 
+          m[12] * m[3] * m[5];
+
+  inv[14] = -m[0]  * m[5] * m[14] + 
+          m[0]  * m[6] * m[13] + 
+          m[4]  * m[1] * m[14] - 
+          m[4]  * m[2] * m[13] - 
+          m[12] * m[1] * m[6] + 
+          m[12] * m[2] * m[5];
+
+  inv[3] = -m[1] * m[6] * m[11] + 
+          m[1] * m[7] * m[10] + 
+          m[5] * m[2] * m[11] - 
+          m[5] * m[3] * m[10] - 
+          m[9] * m[2] * m[7] + 
+          m[9] * m[3] * m[6];
+
+  inv[7] = m[0] * m[6] * m[11] - 
+          m[0] * m[7] * m[10] - 
+          m[4] * m[2] * m[11] + 
+          m[4] * m[3] * m[10] + 
+          m[8] * m[2] * m[7] - 
+          m[8] * m[3] * m[6];
+
+  inv[11] = -m[0] * m[5] * m[11] + 
+          m[0] * m[7] * m[9] + 
+          m[4] * m[1] * m[11] - 
+          m[4] * m[3] * m[9] - 
+          m[8] * m[1] * m[7] + 
+          m[8] * m[3] * m[5];
+
+  inv[15] = m[0] * m[5] * m[10] - 
+          m[0] * m[6] * m[9] - 
+          m[4] * m[1] * m[10] + 
+          m[4] * m[2] * m[9] + 
+          m[8] * m[1] * m[6] - 
+          m[8] * m[2] * m[5];
+
+  var det = m[0] * inv[0] + m[1] * inv[4] + m[2] * inv[8] + m[3] * inv[12];
+
+  if (det === 0) {
+    this._values = m.map(v => NaN);
+    this._is2D = false;
+    return;
+  }
+  
+  this._values = inv.map(v => v/det);
```

Comment:
Instead of making a copy/spare array here, could you divide `inv` in place?

Reply:
good call.